src/main/java/com/atlassian/db/replica/internal/SqlQuery.java

```diff
@@ -4,24 +4,54 @@
     private static final int SELECT_FOR_UPDATE_SUFFIX_LIMIT = 100;
 
     private final String sql;
+    private final boolean compatibleWithPreviousVersion;
 
-    public SqlQuery(String sql) {
+    public SqlQuery(String sql, boolean compatibleWithPreviousVersion) {
```

Comment:
So `sql` is nullable? When?
We should consider `@Nullable`.

Reply:
We pass it from RouteDecisionBuilder builder.
![Zrzut ekranu 2021-04-19 o 12 00 41](https://user-images.githubusercontent.com/3447570/115218464-df1c9480-a106-11eb-892c-a8b9de080de0.png)

Reply:
`new SqlQuery(decisionBuilder.getSql())` 
A `SqlQuery`, which doesn't have the query, doesn't make sense in the domain. 🤔 It shouldn't be created in case of `decisionBuilder.getSql() == null`.

Reply:
https://github.com/atlassian-labs/db-replica/pull/81